### DISCUSSION TITLE
How to properly use Validate() with DateTime.ParseExact  ?

### DISCUSSION THREAD

**Discussion by ILTCED** [JoshClose/CsvHelper#2046] [Q&A]
Hello I am trying to validate DateTime format using the Validate function but documentation doesn't cover DateTime.ParseExact() method.

I get an error that says cannot convert 'CsvHelper.ValidateArgs' to 'System.ReadOnlySpan<char>'

` public class foo
        {
            public DateTime? date { get; set; }
        }`

` Map(m => m.date).Index(0).Validate(args => DateTime.ParseExact(args, "MM/dd/yyyy", CultureInfo.InvariantCulture));`

**Comment by JoshClose** [comment]
`args.Field` 

All of the delegates changes to an args variable with properties on them. You weren't able to tell what the `(string, string, int, string)` (example) was before.

**Reply by ILTCED** [reply]
Hi Thanks for getting back to me.
I've also tried:
`Map(m => m.date).Index(0).Validate(args => DateTime.ParseExact(args.Field, "MM/dd/yyyy", CultureInfo.InvariantCulture));`

That throws an **Cannot Implicitly convert type 'System.DateTime' to 'bool'**
It also throws **'Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type'**

**Reply by JoshClose** [reply]
The `Validate` `delegate` expects you to return a `boolean`. `true` if it's valid, `false` if it's not valid.

Do you want to validate or are you just trying to get the `DateTime` to parse in a specific format?

**Reply by ILTCED** [reply]
I would like to validate the field to ensure that the row entry is the correct 'MM/dd/yyyy' format, then throw an error message. So yes, I want to validate.
